heron/common/src/cpp/config/topology-config-vars.cpp

```diff
@@ -39,6 +39,11 @@ const sp_string TopologyConfigVars::TOPOLOGY_ENABLE_MESSAGE_TIMEOUTS =
 const sp_string TopologyConfigVars::TOPOLOGY_ENABLE_ACKING = "topology.acking";
 const sp_string TopologyConfigVars::TOPOLOGY_CONTAINER_CPU_REQUESTED = "topology.container.cpu";
 const sp_string TopologyConfigVars::TOPOLOGY_CONTAINER_RAM_REQUESTED = "topology.container.ram";
+const sp_string TopologyConfigVars::TOPOLOGY_STATEFUL_ENABLED = "topology.stateful.enabled";
+const sp_string TopologyConfigVars::TOPOLOGY_STATEFUL_CHECKPOINT_INTERVAL_SECONDS =
+                            "topology.stateful.checkpoint.interval.seconds";
+const sp_string TopologyConfigVars::TOPOLOGY_STATEFUL_START_CLEAN = "topology.stateful.start.clean";
+const sp_string TopologyConfigVars::TOPOLOGY_EXACTLYONCE_ENABLED = "topology.exactlyonce.enabled";
```

Comment:
This PR is adding support for a boolean for exactly once, which I don't think we should do, so it is relevant. Doing so will create a poor user experience/design where the right combination of isExactlyOnce and acking needs to be set to for people to get what they want, which is exactly once, at most once or at least once, which is how users think about semantics. Can we design this properly and take an enum? What's the reasoning to not do so now? The enum can only have ExactlyOnce for now if you don't want to fix the other parts. If we did that we at least wouldn't be introducing a new config API that should be deprecated upon launch.